### PR TITLE
ncm-opennebula: Include VNM configuration support for OVS/VXLAN

### LINKED ISSUE #1014: ncm-opennebula: Detect resource capabilities from aii methods
Try to detect the cloud resource capabilities from AII methods before any change to warn users about any resource incompatibility. 
At this moment if the resource is not available or incompatible with the VM requirements the code just sends a generic error message.

To include this feature #1009 should be tested/merged first.

### LINKED ISSUE #1021: ncm-opennebula: include OpenNebulaNetwork.conf OVS support
Some of the Open vSwitch features and rules on the hyps :
 * VLAN trunking
 * Mac-spoofing
 * IP hijacking
...

could be configured through `/var/lib/one/remotes/vnm/OpenNebulaNetwork.conf`   file. Quattor should be able to setup this file using the ONE schema.

### LINKED ISSUE #1111: ncm-opennebula: include VXLAN vnet options
Some VXLAN options are missing from the VNET schema like:
 * `PHYDEV`   the name of the physical network device that will be attached to the bridge **(mandatory)**
 * `MTU`   The MTU for the tagged interface and bridge (optional)

More info:
https://docs.opennebula.org/5.2/deployment/open_cloud_networking_setup/vxlan.html

### LINKED ISSUE #1112: ncm-opennebula: hypervisors are not removed from the db
This issue happens when some hypervisors are removed from the quattor hyp list, the component does not remove them from the db and submits this message:

>[DEBUG] We cannot remove this kvm host. There are still running VMs: hyp0xxxx

Even if there are not running VMs on the hyp. The workaround at this moment is to remove the hyps manually.

### LINKED ISSUE #1114: ncm-opennebula: Add  user/group support for cloud resources
This feature is required by external users and groups, to change the cloud resources managed by quattor like:
 * VNETs
 * Datastores
 to a different user and group instead of oneadmin

### PR DESCRIPTION
**Pull request by alvarosimon**
**NOTE:** This PR requires `Net::OpenNebula 0.311.0` 
 * Fix #1021: Include OpenNebulaNetwork.conf OVS support
 * Fix #1014: Detect resource capabilities from aii methods
 * Fix #1111: Include VXLAN vnet options
 * Fix #1114: Add user/group support for cloud resources
 * Fix #1112: hypervisors are not removed from the db

### CODE REVIEW

File: ncm-opennebula/src/main/perl/OpenNebula/Network.pm [hunk 1]
```diff
@@ -173,11 +173,9 @@ sub remove_vn_ars
         } else {
              $self->error("Unable to remove AR id: $arid from vnet: $vnet");
         }
-    } elsif ($arinfo) {
-        $self->error("Unable to remove AR. QUATTOR flag not found within AR: $arinfo");
     } else {
         $self->warn(1, "Unable to remove AR. ",
-            "AR template is not available from vnet: $vnet: ", $ardata->{ar});
+            "AR is not available or QUATTOR flag si not set vnet: $vnet: ", $ardata->{ar});
```
Comment (stdweird) [inline ncm-opennebula/src/main/perl/OpenNebula/Network.pm]: `si` -> `is`

**Comment by alvarosimon** [comment]
Test this please

**Comment by alvarosimon** [comment]
Test this please

File: ncm-opennebula/src/main/pan/components/opennebula/schema.pan [hunk 2]
```diff
@@ -619,6 +619,24 @@ type opennebula_kvmrc = {
 };
 
 @documentation{
+Type that sets the OpenNebula
+VNM network configuration file on the nodes
```
Comment (stdweird) [inline ncm-opennebula/src/main/pan/components/opennebula/schema.pan]: what does `VNM` stand for?

**Comment by stdweird** [comment]
@alvarosimon minor remarks

File: ncm-opennebula/src/main/pan/components/opennebula/schema.pan (hunk 2 above)
Comment (alvarosimon) [inline ncm-opennebula/src/main/pan/components/opennebula/schema.pan]: Virtual Network Manager, should we include the acronym?

Comment (stdweird) [inline ncm-opennebula/src/main/pan/components/opennebula/schema.pan]: yes please

**Comment by alvarosimon** [comment]
@stdweird done!

**Comment by alvarosimon** [comment]
Test this please

**Comment by alvarosimon** [comment]
Test fails due to:

> Net::OpenNebula version v0.311.0 required--this is only version v0.310.0 at /var/lib/jenkins/workspace/NCM_Components-pull_requests/ncm-opennebula/target/lib/perl/NCM/Component/opennebula.pm line 143.

**Comment by stdweird** [comment]
retest this please

File: ncm-opennebula/src/main/pan/components/opennebula/schema.pan [hunk 3]
```diff
@@ -267,6 +268,25 @@ function is_consistent_datastore = {
...
+        if (!exists(vn['phydev'])) {
+            error("VXLAN vnet requires 'phydev' value to attach a bridge");
+        };
+    # if not the bridge is mandatory
+    } else {
+        if (!exists(vn['bridge'])) {
+            error(format("vnet with 'vn_mad' '%s' requires a 'bridge' value", vn['vn_mad']));
+        };
+    };
+    return(true);
```
Comment (stdweird) [inline ncm-opennebula/src/main/pan/components/opennebula/schema.pan]: no return needed, just `true;`

File: ncm-opennebula/src/main/perl/opennebula.pm [hunk 4]
```diff
@@ -233,6 +235,10 @@ sub create_or_update_something
         # resource is already there and we can modify it
         $new = $self->update_something($one, $type, $name, $template, $data);
     }
+    # Change resource permissions
+    if($new and defined($data->{$name}->{permissions})) {
```
Comment (stdweird) [inline ncm-opennebula/src/main/perl/opennebula.pm]: space after `if`

**Comment by stdweird** [comment]
@alvarosimon minor remarks

**Comment by alvarosimon** [comment]
@stdweird remarks addressed

**Pull request merged**